pkg/services/govmomi/pci/device.go

```diff
@@ -76,7 +76,12 @@ func ConstructDeviceSpecs(pciDeviceSpecs []infrav1.PCIDeviceSpec) []types.BaseVi
 	return pciDevices
 }
 
-func createBackingInfo(spec infrav1.PCIDeviceSpec) *types.VirtualPCIPassthroughDynamicBackingInfo {
+func createBackingInfo(spec infrav1.PCIDeviceSpec) types.BaseVirtualDeviceBackingInfo {
+	if spec.VGPUProfile != "" {
+		return &types.VirtualPCIPassthroughVmiopBackingInfo{
+			Vgpu: spec.VGPUProfile,
```

Comment:
```suggestion
			VGPU: spec.VGPUProfile,
```

Reply:
`Vgpu` is the name of the key in the `VirtualPCIPassthroughVmiopBackingInfo` struct directly from govmomi in this case.